src/Components/v2/Carousel.tsx

```diff
@@ -17,12 +19,32 @@ interface Props {
   renderLeftArrow?: Arrow
   renderRightArrow?: Arrow
   onArrowClick?: () => void
+  lastSlideVisible?: boolean
+  firstSlideVisible?: boolean
+  afterChange?: (currentIndex: number) => void
+}
+
+interface State {
+  lastSlideVisible: boolean
```

Comment:
This feels a tad indirect- namely, that the concept of a first or last slide being visible, has an affect on the corresponding arrow. Why not just have the state object be:

```
interface State {
  canScrollLeft: boolean
  canScrollRight: boolean
}
```

Then, those can be set accordingly. That might make it easier at quick glance to figure out the behavior.